xdsl/dialects/llvm.py

```diff
@@ -95,6 +97,66 @@ def is_typed(self):
         return not isinstance(self.type, NoneAttr)
 
 
+@irdl_op_definition
+class GEPOp(Operation):
+    name = "llvm.getelementptr"
+
+    ptr: Annotated[Operand, LLVMPointerType]
+
+    ssa_indices: Annotated[VarOperand, IntegerType]
+
+    elem_type: OptOpAttr[Attribute]
+
+    rawConstantIndices: OpAttr[DenseArrayBase]
+
+    inbounds: OptOpAttr[UnitAttr]
+
+    result: Annotated[OpResult, LLVMPointerType]
+
+    @staticmethod
+    def get(ptr: SSAValue | Operation,
+            result_type: Attribute
+            | LLVMPointerType = LLVMPointerType.opaque(),
+            indices: list[int] | None = None, #Here we are assuming the indices follow the MLIR standard (min int where the SSA value should be used)
+            ssa_indices: list[SSAValue | Operation] | None = None,
+            inbounds: bool = False,
+            pointee_type: Attribute | None = None):
+
+        if indices is None:
+            raise ValueError('llvm.getelementptr must have a pointer.')
+
+        indices_attr = DenseArrayBase.create_dense_int_or_index(i32, indices)
+
+        # construct default mutable argument here:
+        if ssa_indices is None:
+            ssa_indices = []
+
+        # convert a potential Operation into an SSAValue
+        ptr = SSAValue.get(ptr)
+
+        if not isinstance(result_type, LLVMPointerType):
+            raise ValueError('Result type must be a pointer.')
+
+        if not ptr.typ.is_typed():  #type: ignore
```

Comment:
could you please remove the `#type: ignore`? Feel free to DM me for help debugging the type issues